File: tasks/2024/round3/bracketswap.py
#!/usr/bin/env python3
# @check-accepted: examples Nsmall Nmedium Nlarge

input()
s = list(input().strip())

mn, sum = 0, 0
for i in s:
    sum += 1 if i == '(' else -1
    mn = min(mn, sum)

ans = []

l, r = 0, len(s)-1
while l<r and mn < 0:
    if s[l]=='(':
        l += 1
    elif s[r]==')':
        r -= 1
    else:
        assert s[l]==')' and s[r]=='('
        ans.append((l, r))
        s[l], s[r] = s[r], s[l]
        mn+=2

print(len(ans))
print(*[f"{x[0]} {x[1]}" for x in ans], sep='\n')

File: tasks/2024/round3/bracketswap.cpp
// @check-accepted: examples Nsmall Nmedium Nlarge
#include <iostream>
#include <vector>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    int n;
    cin >> n;
    string s;
    cin >> s;
    int bal = 0, mn = 0;
    for (char c : s) {
        bal += c == '(' ? 1 : -1;
        mn = min(mn, bal);
    }
    vector<pair<int, int>> ans;
    int l = 0, r = n - 1;
    while (mn < 0) {
        while (s[l] == '(') l++;
        while (s[r] == ')') r--;
        ans.push_back({l, r});
        swap(s[l], s[r]);
        mn += 2;
    }
    cout << ans.size() << "\n";
    for (auto& [l, r] : ans) {
        cout << l << " " << r << "\n";
    }
}
